frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -238,3 +239,15 @@ export const usePrometheusPoll: UsePrometheusPoll = (options) => {
   // unify order with the rest of API
   return [result[0], !result[2], result[1]];
 };
+
+/**
+ * YAML editor field component
+ * @param label - (optional) label that will show on the side bar
+ * @param model - (optional) resource model
+ * @param schema - (optional) resource schema
+ * @param showSamples - flag for sample viewing in sidebar
+ * @param onSave - callback for saving the yaml
+ * @param value - yaml value
+ * @param onChange - callback for yaml changes
+ */
+export const YAMLEditorField: React.FC<YAMLEditorFieldProps> = require("@console/shared/src/components/formik-fields/YAMLEditorField").default;
```

Comment:
lets rename it to `YAMLEditorWithSidebar` because `YAMLEditorField` isnt very obvious on why would I use this component instead of just any other `YAMLEditor` (like PF one)

Reply:
I suggest ResourceYAMLEditorField.
The Resource part indicates that this isn't a generic yaml editor like the PF one, but specific for a Open shift resource and thus comes with additional data on the resource, displayed in a sidebar.
The Field part indicates that it's different than [EditYAML](https://github.com/openshift/console/blob/master/frontend/public/components/edit-yaml.jsx) in that the use case for it is meant to be as part as a form, so it doesn't include the create/save button

Reply:
Sounds good, but I'd remove the `Fiield` soo... `ResourceYAMLEditor`